### PR TITLE
docs, community: Runhouse integration update

### PR DESCRIPTION
**Pull request by BelSasha**
**Description:** Updated the Runhouse integration classes and usage examples (notebooks in the docs), following the latest Runhouse changes and new capabilities.

### CODE REVIEW

File: libs/community/langchain_community/embeddings/self_hosted.py [hunk 1]
```diff
@@ -29,47 +43,51 @@ class SelfHostedEmbeddings(SelfHostedPipeline, Embeddings):
...
-                model_reqs=["./", "torch", "transformers"],
-            )
+                def load_embedding_model(self):
+                    ... load_embedding_model implementation here ...
+
+                def embed_documents(self, *args: Any, **kwargs: Any) -> List[List[float]]:
+                    .. embed_documents implementation here ...
+
+            gpu = rh.cluster(name='rh-a10x', instance_type='g5.4xlarge', provider='aws')
+            gpu.run(commands=["pip install langchain"])
```
Comment (jlewitt1) [inline libs/community/langchain_community/embeddings/self_hosted.py:59]: assume is this bc we need langchain installed on the cluster before we install the env?

File: docs/docs/integrations/llms/runhouse.ipynb [hunk 2]
```diff
@@ -72,250 +64,155 @@
...
-   "cell_type": "code",
-   "execution_count": 36,
-   "id": "feb8da8e",
+   "execution_count": 20,
+   "id": "1528e70f",
    "metadata": {},
    "outputs": [
     {
-     "name": "stdout",
+     "name": "stderr",
```
Comment (jlewitt1) [inline docs/docs/integrations/llms/runhouse.ipynb:184]: how come this is stderr now?

File: docs/docs/integrations/llms/runhouse.ipynb [hunk 3]
```diff
@@ -51,16 +43,16 @@
...
    "id": "06d6866e",
    "metadata": {
     "tags": []
    },
    "outputs": [],
    "source": [
     "# For an on-demand A100 with GCP, Azure, or Lambda\n",
-    "gpu = rh.cluster(name=\"rh-a10x\", instance_type=\"A100:1\", use_spot=False)\n",
-    "\n",
+    "gpu = rh.cluster(name=\"langchain-rh-a10x\", instance_type=\"g5.4xlarge\", provider=\"aws\")\n",
```
Comment (jlewitt1) [inline docs/docs/integrations/llms/runhouse.ipynb:54]: maybe better to keep without specifying a provider? and use the default "cheapest" behavior

File: docs/docs/integrations/llms/runhouse.ipynb [hunk 4]
```diff
@@ -334,7 +231,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.10.12"
+   "version": "3.9.15"
```
Comment (jlewitt1) [inline docs/docs/integrations/llms/runhouse.ipynb:234]: any particular reason for the version downgrade?

File: libs/community/langchain_community/llms/self_hosted.py [hunk 5]
```diff
@@ -224,12 +141,12 @@ def _llm_type(self) -> str:
         return "self_hosted_llm"
 
     def _call(
-        self,
-        prompt: str,
-        stop: Optional[List[str]] = None,
-        run_manager: Optional[CallbackManagerForLLMRun] = None,
-        **kwargs: Any,
+            self,
```
Comment (jlewitt1) [inline libs/community/langchain_community/llms/self_hosted.py:144]: dumb nit, i think need to run the linter

File: libs/community/langchain_community/llms/self_hosted.py [hunk 6]
```diff
@@ -12,57 +11,66 @@
...
+        from transformers import pipeline as hf_pipeline
+
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_id,
+                                                       token=hf_token)
+
+        try:
+            self.model = AutoModel.from_pretrained(self.model_id,
+                                                   token=hf_token)
+        except RuntimeError:
+            self.model = AutoModelForMaskedLM.from_pretrained(self.model_id,
```
Comment (jlewitt1) [inline libs/community/langchain_community/llms/self_hosted.py:35]: Not sure i totally follow - if we can't load as a generic `AutoModel` we will always be able to with `AutoModelForMaskedLM`?

File: libs/community/langchain_community/llms/self_hosted.py [hunk 7]
```diff
@@ -1,8 +1,7 @@
-import importlib.util
 import logging
-import pickle
-from typing import Any, Callable, List, Mapping, Optional
+from typing import Any, List, Mapping, Optional
 
+import runhouse as rh
```
Comment (baskaryan) [inline libs/community/langchain_community/llms/self_hosted.py:4]: needs to be conditional import (within the function that use it)

**Comment by efriis** [comment]
closing as stale / no response

**Pull request closed without merging**